### PR TITLE
add Cody role based access control docs

### PR DESCRIPTION
**Pull request by emidoots**
Corresponding docs for https://github.com/sourcegraph/sourcegraph/pull/58831

### CODE REVIEW

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 1]
```diff
@@ -79,7 +79,36 @@ To turn Cody off:
 
 ## Enable Cody only for some users
 
-To enable Cody only for some users, for example, when rolling out a Cody POC, follow all the steps mentioned in [Enabling Cody on your Sourcegraph instance](#enable-cody-on-your-sourcegraph-instance). Then, use the feature flag `cody` to turn Cody on selectively for some users. To do so:
+To enable Cody only for some users, for example, when rolling out a Cody POC, follow all the steps mentioned in [Enabling Cody on your Sourcegraph instance](#enable-cody-on-your-sourcegraph-instance). Then, do the following:
+
+<details>
```
Comment (MaedahBatool) [inline docs/cody/clients/enable-cody-enterprise.mdx:84]:
```suggestion

```
These HTML tags will not work. We will soon get a `toggle` component that will make it easier. Till then we can use an H3. :)

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 2]
```diff
@@ -79,7 +79,36 @@ To turn Cody off:
...
+<summary>Sourcegraph 5.3+</summary>
```
Comment (MaedahBatool) [inline docs/cody/clients/enable-cody-enterprise.mdx:85]:
```suggestion
### Sourcegraph 5.3+
```

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 3]
```diff
@@ -79,7 +79,36 @@ To turn Cody off:
...
+
+```json
+    {
+      // [...]
+      "cody.enabled": true,
+      // Make sure cody.restrictUsersFeatureFlag is not in your configuration! If it is, remove it.
+    }
+```
+
+**Note:** Ensure `cody.restrictUsersFeatureFlag` is NOT in your site configuration. If it is, remove it or else the old feature-flag approach from Sourcegraph 5.2 and earlier will be used.
```
Comment (MaedahBatool) [inline docs/cody/clients/enable-cody-enterprise.mdx:99]:
```suggestion
<Callout type="note"> Ensure `cody.restrictUsersFeatureFlag` is **not** in your site configuration. If it is, remove it or else the old feature-flag approach from Sourcegraph 5.2 and earlier will be used.</Callout>
```

We have a component called `Callout` for `note`, `info`, `tip`, and `warning` types.

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 4]
```diff
@@ -79,7 +79,36 @@ To turn Cody off:
...
+
+Next, go to **Site admin > Users & Auth > Roles** (`/site-admin/roles`) on your instance. On that page, you can:
+
+* Control whether users _by default_ have access to Cody (expand `User [System]` and toggle **Cody** > **Access** as desired)
```
Comment (MaedahBatool) [inline docs/cody/clients/enable-cody-enterprise.mdx:103]:
```suggestion
- Control whether users _by default_ have access to Cody (expand `User [System]` and toggle **Cody** > **Access** as desired)
```

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 5]
```diff
@@ -79,7 +79,36 @@ To turn Cody off:
...
+* Control whether groups of users have access to Cody (`+Create role` and enable the **Cody** > **Access** toggle as desired)
```
Comment (MaedahBatool) [inline docs/cody/clients/enable-cody-enterprise.mdx:104]:
```suggestion
- Control whether groups of users have access to Cody (`+Create role` and enable the **Cody** > **Access** toggle as desired)
```

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 6]
```diff
@@ -79,7 +79,36 @@ To turn Cody off:
...
+
+</details>
```
Comment (MaedahBatool) [inline docs/cody/clients/enable-cody-enterprise.mdx:106]:
```suggestion

```

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 7]
```diff
@@ -79,7 +79,36 @@ To turn Cody off:
...
+
+<details>
```
Comment (MaedahBatool) [inline docs/cody/clients/enable-cody-enterprise.mdx:108]:
```suggestion

```

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 8]
```diff
@@ -79,7 +79,36 @@ To turn Cody off:
...
+<summary>Sourcegraph 5.2 and earlier</summary>
```
Comment (MaedahBatool) [inline docs/cody/clients/enable-cody-enterprise.mdx:109]:
```suggestion
### Sourcegraph 5.2 and earlier
```

File: docs/cody/clients/enable-cody-enterprise.mdx [hunk 9]
```diff
@@ -102,6 +131,8 @@ To enable Cody only for some users, for example, when rolling out a Cody POC, fo
   src="https://user-images.githubusercontent.com/25070988/235454594-9f1a6b27-6882-44d9-be32-258d6c244880.png"
 />
 
+</summary>
```
Comment (MaedahBatool) [inline docs/cody/clients/enable-cody-enterprise.mdx:134]:
```suggestion

```

**Review by MaedahBatool — CHANGES_REQUESTED** [review]
Left some comments basically around structure. Rest looks good. Thanks for contributing. :)

**Comment by slimsag** [comment]
@MaedahBatool done!

**Review by MaedahBatool — APPROVED** [review]

**Pull request merged**